src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,194 @@
+                    .of(getCarWithManufacturerFromResultSet(preparedStatement.executeQuery()));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get field from car table with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model AS car_model, "
+                + "m.id AS manufacturer_id, m.name AS manufacturer_name, "
+                + "m.country AS manufacturer_country "
+                + "FROM car c "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarWithManufacturerFromResultSet(resultSet));
+            }
+            for (Car car : cars) {
+                car.setDrivers(getDriversForCar(car));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from car table", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE car SET model = ?, manufacturer_id = ? "
+                + "WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement
+                         = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car table with car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE car SET is_deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.execute();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete field with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```

Comment:
It`s possible to break down same logic in getAllByDriver and getAll in separate logic to prevent typing same code multiple times.

Reply:
good point